packages/valory/skills/price_estimation_abci/models.py

```diff
@@ -64,28 +61,11 @@ class SharedState(BaseSharedState):
     abci_app_cls = PriceAggregationAbciApp
 
 
-# TODO remove this workaround when the types get fixed in `open-autonomy`
-class FixedApiSpecs(ApiSpecs):
-    """A model that wraps APIs to get cryptocurrency prices."""
-
-    def __init__(self, *args: Any, **kwargs: Any) -> None:
-        """Initialize ApiSpecsModel."""
-        self.url: str = self._ensure("url", kwargs, str)
-        self.api_id: str = self._ensure("api_id", kwargs, str)
-        self.method: str = self._ensure("method", kwargs, str)
-        self.headers = self._ensure("headers", kwargs, List[OrderedDict[str, str]])
-        self.parameters = self._ensure("parameters", kwargs, List[List[str]])
-        self.response_info = ResponseInfo.from_json_dict(kwargs)
-        self.retries_info = RetriesInfo.from_json_dict(kwargs)
-        super(Model, self).__init__(*args, **kwargs)  # pylint: disable=bad-super-call
-        self._frozen = True
-
-
-class RandomnessApi(FixedApiSpecs):
+class RandomnessApi(ApiSpecs):
```

Comment:
A refactor that I forgot to perform on #48. The `open-autonomy v0.9.0` now fixes the types for the `parameters` and `headers`. However, the dicts are not overridable, so I had to add another hack on e8cb8205c384b3caabac4f5181266a9fd0f80663. We should definitely address this issue at some point:
1. It is really not ideal to specify all the keys in advance.
2. What if we need some agents to have different keys? Then workarounds like the following need to be performed:
https://github.com/valory-xyz/price-oracle/blob/df14321efcce4fa36e9d5c1afc86d41780bc4423/packages/valory/skills/price_estimation_abci/models.py#L79-L81

Reply:
cc @angrybayblade